src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,124 @@
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(CREATE_BOOK,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int row = statement.executeUpdate();
+            if (row < 1) {
+                throw new DataProcessingException("Expected to insert at least 1 row, but inserted "
+                        + row + " rows");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't save a book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BOOK_BY_ID)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = createBookObject(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_BOOK)) {
+            List<Book> allBooks = new ArrayList<>();
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book bookObject = createBookObject(resultSet);
+                allBooks.add(bookObject);
+            }
+            return allBooks;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all books from db ", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_BOOK)) {
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            statement.setObject(3, book.getId());
+            int row = statement.executeUpdate();
+            if (row < 1) {
+                throw new DataProcessingException("Expected to update row "
+                        + row + " row");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book " + book, e);
```

Comment:
Use informative messages for exceptions. For example, include the book id when an error occurs while updating a book.